cpp/src/strings/replace/multi_re.cu

```diff
@@ -97,13 +97,12 @@ struct replace_multi_regex_fn {
       }
       // all the ranges have been updated from each regex match;
       // look for any that match at this character position (ch_pos)
-      auto itr = thrust::find_if(
-        thrust::seq, d_ranges, d_ranges + number_of_patterns, [ch_pos] __device__(auto range) {
+      auto itr =
+        thrust::find_if(thrust::seq, d_ranges, d_ranges + number_of_patterns, [ch_pos](auto range) {
```

Comment:
What's the purpose of removing the __device__ tag?  I'm surprised you can do this.

Reply:
I was told that when using `thrust::seq` inside a device function, the lambda does not need to have the `__device__` declaration. I'm guessing it is because code does not launch a kernel. And because everything is inlined.

Reply:
Lambdas defined inside device code don't need `__device__`. It's only when you define a lambda in host code that you need the decorator.